text/0117-typo-setting-api.md

```diff
@@ -0,0 +1,295 @@
+
+- 🔴 Sending invalid integer values for the `minWordSizeForTypos` object fields returns an [invalid_typo_min_word_size_for_typos](0061-error-format-and-definitions.md#invalid_typo_min_word_size_for_typos) error.
+
+##### 3.2.2.2.3. Lazy Index Creation
+
+If the requested `index_uid` does not exist, and the authorization layer allows it (See [3.2.2.2.2.1. Async Errors](#322221-async-errors)), Meilisearch will create the index when the related asynchronous task resource is executed. See [3.2.2.2.1. Response Definition](#32221-response-definition).
+
+#### 3.2.2.3. `DELETE`- `indexes/:index_uid/settings/typo`
+
+Allow resetting the typo tolerance feature to the default for an index.
+
+##### 3.2.2.3.1. Response Definition
+
+When the request is in a successful state, Meilisearch returns the HTTP code `202 Accepted`. The response's content is the summarized representation of the received asynchronous task.
+
+See [Summarized `task` Object for `202 Accepted`](0060-tasks-api.md#summarized-task-object-for-202-accepted).
+
+##### 3.2.2.3.2. Errors
+
+###### 3.2.2.3.2.1. Asynchronous Index Not Found Error
+
+- 🔴 If the requested `index_uid` does not exist, the API returns an [index_not_found](0061-error-format-and-definitions.md#index_not_found) error in the related async `task` resource. See [3.2.2.3.1. Response Definition](#32231-response-definition).
+
+### 3.2.3. General Errors
+
+These errors apply to all endpoints described here.
+
+#### 3.2.3.1. Auth Errors
+
+The auth layer can return the following errors if Meilisearch is secured (a master-key is defined).
+
+- 🔴 Accessing this route without the `Authorization` header returns a [missing_authorization_header](0061-error-format-and-definitions.md#missing_authorization_header) error.
+- 🔴 Accessing this route with a key that does not have permissions (i.e. other than the master-key) returns an [invalid_api_key](0061-error-format-and-definitions.md#invalid_api_key) error.
+
+## 2. Technical Details
+
+### 2.1. Typos Calculation
+
+The size of a query term triggers the typo tolerance feature.
+
+By default, a query term made of `5` characters can tolerate `1` typo, while a query term made of `9` characters can tolerate `2` typos.
+
+If a query term can be derived from a distance of one or two typos to be found in a document, it will be selected as a candidate. The number of typos found has an impact on the relevancy.
+
+#### 2.1.1 Typo On The First Character
+
+A query term having the first character considered as a typo will not count as `1` typo but as `2` typos.
+
+It permits to drastically reduce the search space in this case, thus speed-up the search response.
+
+#### 2.1.2. Clamping Typo On Concatened Query Terms
+
+The concatenation of two query terms is equal to 1 typo.
+
+e.g. If `Le tableau` is typed,  the concatened string `Letableau` made of `9` letters can only have `1` typo instead of `2` since `1` typo is already counted by the concatenation.
```

Comment:
Will this rule apply if the query was `Le-tableau` or `Le,tableau`?

Reply:
Yes, it will as `-` and `,` are stop-words like ` `.

Reply:
Thanks, @maryamsulemani97 and @Kerollmops ! I will precise it in the specification.